lib/auth/Vault.ts

```diff
@@ -356,6 +358,35 @@ export default class Vault {
             });
     }
 
+    /**
+     * A getter for account canonical IDs given a list of account IDs
+     * @param accountIds - list of account IDs
+     * @param log - log object
+     * @param callback - callback function
+     * @returns callback with either error or an object from Vault
+     * containing canonicalID and display name for each account ID
+     */
+    getCanonicalIdsByAccountIds(
+        accountIds: string[],
+        log: RequestLogger,
+        callback: GenericCallback<AccountCannonicalInfo[]>,
+    ) {
+        log.addDefaultFields({ accountIds, method: 'getCanonicalIdsByAccountIds' });
+        log.trace('getting canonicalIDs from Vault based on accountIDs');
+        const options = {
+            reqUid: log.getSerializedUids(),
+            logger: log,
+        };
+        this.client.getCanonicalIdsByAccountIds(accountIds, options,
+            (err: Error | null, res?: AccountCannonicalInfoResults) => {
+                if (err) {
+                    log.error('received error message from vault', { error: err });
```

Comment:
```suggestion
                    log.debug('received error message from vault', { error: err });
```

Usually I prefer debug logs in case of errors in "low-level" methods like that: this function alone does not know if the error is expected or not. Maybe we expect from above that none of the provided ID are found. So better to let the upper layers log with `error`: this will dump the debug log here.

```suggestion
                    log.debug('received error message from vault', { error: err });
```